### PR TITLE
[8.x] Use/exclude specific resource abilities in AuthorizeRequests

### PR DESCRIPTION
**Pull request by ajcastro**
This PR provides a fluent way to set specific authorization model policies in the controller's constructor 
used together with `authorizeResource()` method. 
I believe this feature should be in the core so that everyone will be doing it in a uniform way instead of having different implementations. Feel free to choose if there is more approriate method names.

Example:
```php
class UsersController extends Controller
{
    public function __construct() 
    {
        $this->onlyResourceAbilities(['create', 'update', 'delete'])
             ->authorizeResource(User::class);
    }
}
```

```php
class UsersController extends Controller
{
    public function __construct() 
    {
        $this->excludeResourceAbilities(['viewAny', 'view'])
             ->authorizeResource(User::class);
    }
}
```

Extra feature: Set resource abilities in constructor.

```php
class UsersController extends Controller
{
    public function __construct() 
    {
        $this->setResouceAbilities([
            'index' => 'viewAny',
            'show' => 'view',
            'create' => 'create',
            'store' => 'create',
            'edit' => 'update',
            'update' => 'update',
            'destroy' => 'delete',
        ])
        ->authorizeResource(User::class);
    }
}
```

### CODE REVIEW

**Comment by jrmajor** [comment]
Where is `authorizeRequests()` method defined? Did you mean `authorizeResource()`?

Also, this can be easily done by overwriting `resourceAbilityMap()` method.

File: src/Illuminate/Foundation/Auth/Access/AuthorizesRequests.php [hunk 1]
```diff
@@ -97,22 +112,60 @@ public function authorizeResource($model, $parameter = null, array $options = []
...
+        $this->resourceAbilities = $resourceAbilities;
+
+        return $this;
+    }
+
+    /**
+     * Use only the given abilities.
+     *
+     * @param string|array $abilities
+     * @return this
```
Comment (jrmajor) [inline src/Illuminate/Foundation/Auth/Access/AuthorizesRequests.php:131]:
```suggestion
     * @param  string|array  $abilities
     * @return $this
```

The same goes for other docblocks.

**Review by jrmajor — CHANGES_REQUESTED** [review]

File: src/Illuminate/Foundation/Auth/Access/AuthorizesRequests.php (hunk 1 above)
Comment (ajcastro) [reply src/Illuminate/Foundation/Auth/Access/AuthorizesRequests.php:131]: Yes I mean `authorizeResource`. Sorry about the typo

**Comment by ajcastro** [comment]
@jrmajor Yes, I mean `authorizeResource()`. Sorry about the typo. I updated it now.
Maybe you're right that it's maybe better to just override the method. Anyway, this is just some like syntactical sugar.

**Comment by ajcastro** [comment]
I guess not everyone actually needs this solution and I think probably others may have a different approach. So it's fine if this should be closed.

**Pull request closed without merging**